swarmrl/observables/subdivided_vision_cones.py

```diff
@@ -0,0 +1,165 @@
+"""
+Computes vision cone(s).
+"""
+from abc import ABC
+from typing import List
+
+import numpy as np
+
+from swarmrl.models.interaction_model import Colloid
+from swarmrl.observables.observable import Observable
+from swarmrl.tasks.task import Task
+
+
+class SubdividedVisionCones(Observable, ABC, Task):
+    """
+    The vision cone acts like a camera for the particles. It can either output all
+    colloids within its view, a function of the distances between one colloid and all
+    other colloids, or the normalised distance to the source.
+    """
+
+    def __init__(
+        self, vision_range: float, vision_half_angle: float, n_cones: int, radii=[]
+    ):
+        """
+        Constructor for the observable.
+
+        Parameters
+        ----------
+        vision_range : float
+                How far the particles can see.
+        vision_half_angle : float
+                the half width of the field of view in radiant units
+        n_cones: int
+                In how many cone is the field of view subdivided
+        radii: list
+                List of the radii of the colloids in the experiment,
+                ordered by the ids of the colloids
+        """
+        self.vision_range = vision_range
+        self.vision_half_angle = vision_half_angle
+        self.n_cones = n_cones
+        self.radii = radii
+        self.types = []
+
+    def initialize(self, colloids: List[Colloid]):
```

Comment:
if it does nothing (and the parent class fcn does also just pass) omit this function